plonky2/src/batch_fri/oracle.rs

```diff
@@ -165,6 +173,16 @@ impl<F: RichField + Extendable<D>, C: GenericConfig<D, F = F>, const D: usize>
                 quotient.coeffs.push(F::Extension::ZERO); // pad back to power of two
                 alpha.shift_poly(&mut final_poly);
                 final_poly += quotient;
```

Comment:
It will be shifted at the next step, won't it (since at the next step, final_poly = quotient, and we apply `alpha.shift_poly(&mut final_poly);`)? Or am I missing something?

Reply:
Hmmm... I am confused. First of all, the having the zk-clause in each `{point, polynomial}` instance suggests as if we had the possibility of several mask polynomials `R(X)`. I know it is activated only in the first instance `idx = 0`, but still confusing.
 Anyway, assuming that the loop starts with instance `idx = 0`, then in this step `final_poly` is still empty and we add `R(X)` to the quotient 
```
Q(X) = (p_0(X) - v_0)/(X - z) + alpha * (p_1(X) - v_1)/(X - z) + ... ,
```
resulting in
```
R(X) + (p_0(X) - v_0)/(X - z) + alpha * (p_1(X) - v_1)/(X - z) + ... ,
```
with two terms belonging to the `alpha^0`-weight.